File: app/src/main/java/com/stupidtree/cloudliter/data/model/RelationEvent.java
package com.stupidtree.cloudliter.data.model;

import java.sql.Timestamp;
import java.util.Objects;

public class RelationEvent {
    public enum STATE {REQUESTING, ACCEPTED, REJECTED, DELETE}

    public enum ACTION {ACCEPT, REJECT}

    String id;
    String userId;
    String friendId;
    String otherId;
    String otherAvatar;
    String otherNickname;
    STATE state;
    boolean unread;
    Timestamp createdAt;
    Timestamp updatedAt;

    public String getId() {
        return id;
    }

    public String getUserId() {
        return userId;
    }

    public String getFriendId() {
        return friendId;
    }

    public String getOtherAvatar() {
        return otherAvatar;
    }

    public String getOtherNickname() {
        return otherNickname;
    }

    public STATE getState() {
        return state;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public String getOtherId() {
        return otherId;
    }

    public Timestamp getUpdatedAt() {
        return updatedAt;
    }


    public boolean isUnread() {
        return unread;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RelationEvent that = (RelationEvent) o;
        return Objects.equals(getId(), that.getId()) &&
                Objects.equals(getUserId(), that.getUserId()) &&
                Objects.equals(getFriendId(), that.getFriendId()) &&
                Objects.equals(getOtherAvatar(), that.getOtherAvatar()) &&
                Objects.equals(getOtherNickname(), that.getOtherNickname()) &&
                Objects.equals(isUnread(), that.isUnread()) &&
                getState() == that.getState();
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), getUserId(), getFriendId(), getOtherAvatar(), getOtherNickname(), getState());
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/data/model/RelationEvent.kt
package com.stupidtree.cloudliter.data.model

import java.sql.Timestamp
import java.util.*

class RelationEvent {
    enum class STATE {
        REQUESTING, ACCEPTED, REJECTED, DELETE
    }

    enum class ACTION {
        ACCEPT, REJECT
    }

    var id: String? = null
    var userId: String? = null
    var friendId: String? = null
    var otherId: String? = null
    var otherAvatar: String? = null
    var otherNickname: String? = null
    var state: STATE? = null
    var isUnread = false
    var createdAt: Timestamp? = null
    var updatedAt: Timestamp? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as RelationEvent
        return id == that.id &&
                userId == that.userId &&
                friendId == that.friendId &&
                otherAvatar == that.otherAvatar &&
                otherNickname == that.otherNickname &&
                isUnread == that.isUnread && state == that.state
    }

    override fun hashCode(): Int {
        return Objects.hash(id, userId, friendId, otherAvatar, otherNickname, state)
    }
}